src/triage/experiments/multicore.py

```diff
@@ -2,15 +2,24 @@
 import traceback
 from functools import partial
 from pebble import ProcessPool
+from triage.util.pickling import can_pickle
+from triage.util.db import create_engine
 
 from triage.component.catwalk.utils import Batch
 
 from triage.experiments import ExperimentBase
 
 
 class MultiCoreExperiment(ExperimentBase):
-    def __init__(self, n_processes=1, n_db_processes=1, *args, **kwargs):
-        super(MultiCoreExperiment, self).__init__(*args, **kwargs)
+    def __init__(self, db_engine, n_processes=1, n_db_processes=1, *args, **kwargs):
+        if not can_pickle(db_engine):
+            logging.warning(
+                "Raw, unserializable SQLAlchemy engine passed. "
+                "URL will be used, other options may be lost in multi-process environments"
+            )
+            db_engine = create_engine(db_engine.url)
```

Comment:
I'm still unsure it makes sense to just write a warning to the logs, and replace their engine, carrying through only the database URL.

It seems more user-friendly, to me – for something like this anyway – to raise an exception, so they can resolve it themselves, (most likely by availing themselves of our `create_engine`), and thereby ensure that their engine is properly created with the arguments they want.

Reply:
Yeah, that's true